src/corehaptics.cs

```diff
@@ -0,0 +1,394 @@
+//
+// CoreHaptics C# bindings
+//
+// Authors:
+//	Manuel de la Pena Saenz <mandel@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+
+using System;
+	
+using AVFoundation;
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+namespace CoreHaptics {
+
+#if MONOMAC
+	interface AVAudioSession {}
+#endif
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticEventParameter {
+		[BindAs (typeof (CHHapticEventParameterId))]
+		[Export ("parameterID")]
+		NSString ParameterId { get; }
+
+		[Export ("value")]
+		float Value { get; set; }
+
+		[Export ("initWithParameterID:value:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([BindAs (typeof (CHHapticEventParameterId))] NSString parameterId, float value);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticDynamicParameter {
+		[BindAs (typeof (CHHapticDynamicParameterId))]
+		[Export ("parameterID")]
+		NSString ParameterId { get; }
+
+		[Export ("value")]
+		float Value { get; set; }
+
+		[Export ("relativeTime")]
+		double RelativeTime { get; set; }
+
+		[Export ("initWithParameterID:value:relativeTime:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([BindAs (typeof (CHHapticDynamicParameterId))] NSString parameterId, float value, double time);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticParameterCurveControlPoint {
+		[Export ("relativeTime")]
+		double RelativeTime { get; set; }
+
+		[Export ("value")]
+		float Value { get; set; }
+
+		[Export ("initWithRelativeTime:value:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (double time, float value);
+	}
+
+	[Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface CHHapticParameterCurve {
+		[BindAs (typeof (CHHapticDynamicParameterId))]
+		[Export ("parameterID")]
+		NSString ParameterId { get; }
+
+		[Export ("relativeTime")]
+		double RelativeTime { get; set; }
+
+		[Export ("controlPoints")]
+		CHHapticParameterCurveControlPoint[] ControlPoints { get; }
+
+		[Export ("initWithParameterID:controlPoints:relativeTime:")]
+		[DesignatedInitializer]
+		IntPtr Constructor ([BindAs (typeof (CHHapticDynamicParameterId))]NSString parameterID, CHHapticParameterCurveControlPoint[] controlPoints, double relativeTime);
```

Comment:
`parameterID` -> `parameterId`

Reply:
What does the `[BindAs]` attribute do? Does it map the NSString of parameterID to CHHapticDynamicParameterId somehow? When do we usually use it?